modules-js/react-fleet/src/elements/inputs/FileInput.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import { css } from 'emotion';
+
+import { VISUALLYHIDDEN } from '../../utilities/css';
+
+import { MEDIA_SMALL, OPTIMISTIC_BLUE } from '../../utilities/constants';
+
+import CloseButton from '../buttons/CloseButton';
+
+
+const INPUT_CONTAINER_STYLE = css(`
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+
+  ${MEDIA_SMALL} {
+    flex-wrap: nowrap;
+  }
+  
+  label {
+    margin-bottom: 0.5em;
+
+    ${MEDIA_SMALL} {
+      margin-bottom: 0;
+    }
+  }
+  
+  div {
+    flex-basis: 80%;
+        
+    ${MEDIA_SMALL} {
+      flex-basis: auto;
+      padding-left: 1em;
+    }
+  }
+`);
+
+const LABEL_FOCUSED_STYLE = css({
+  outline: `3px solid ${OPTIMISTIC_BLUE}`,
+  outlineOffset: '1px'
+});
+
+// todo: safari is not ever picking up this styling?
+const FILE_PREVIEW_STYLE = css({
+  position: 'relative',
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis'
+});
+
+const DELETE_BUTTON_STYLE = css({
+  marginLeft: '0.2em'
+});
+
+const DEFAULT_PREVIEW_TEXT = 'No file selected';
+
+
+interface Props {
+  name: string;
+  title: string;
+  fileTypes: string;
+  sizeLimit: fileSize;
+  // parentFileHandler: any;
+}
+
+interface State {
+  selectedFile: any;
+  isFocused: boolean;
+}
+
+interface fileSize {
+  quantity: string | number;
+  unit: string;
+}
+
+export default class FileInput extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      selectedFile: {},
+      isFocused: false
+    }
+  }
+
+  private inputRef = React.createRef<HTMLInputElement>();
+
+  /**
+   * Only accept the file if it doesn’t exceed the specified limit
+   */
+  private checkSize = (file: any) => {
+    const { quantity, unit } = this.props.sizeLimit;
+    const result = formatBytes(file.size);
+
+    if (result.unit === unit && +quantity >= +quantity) {
+      // todo: should this be a modal/notification component?
+      return alert(`File size limit of ${quantity}${unit} exceeded.\n\nYour file is ${result.quantity + result.unit}. Please select a different file.`);
+    }
+
+    this.setState({ selectedFile: file });
+  };
+
+  /**
+   * Ensure file is cleared in the state, as well as in the input element itself
+   */
+  private clearFile = () => {
+    // @ts-ignore
+    this.setState({ selectedFile: {} }, () => this.inputRef.current.value = '');
+  };
+
+  private setFocus = (isFocused: boolean) => {
+    this.setState({ isFocused });
+  };
+
+  private handleChange = () => {
+    // @ts-ignore
+    if (!this.inputRef.current.files.length) {
+      this.clearFile();
+
+    } else {
+      // @ts-ignore
+      this.checkSize(this.inputRef.current.files[0]);
+    }
+  };
+
+  private handleBlur = () => {
+    // todo: parent passes down handler method to collect the file?
+    // todo: or formik will manage?
+    // this.props.parentFileHandler(this.state.selectedFile);
+
+    this.setFocus(false);
```

Comment:
Yeah, it would be worth seeing how this could be more integrated with Formik, perhaps even to the point of validating file sizes through (extensions written on top of) Yup.

Something for a later PR, though.